controllers/harbor/provision.go

```diff
@@ -0,0 +1,307 @@
+}
+
+// getChartMuseumCacheSecret will get a name of k8s secret which stores cache info
+func (harbor *HarborReconciler) getChartMuseumCacheSecret() string {
+	p := harbor.getStorageProperty("chartMuseumSecret")
+	if p == nil {
+		return ""
+	}
+	return p.ToString()
+}
+
+// getStorageSecret will get a name of k8s secret which stores storage info
+func (harbor *HarborReconciler) getStorageSecret() string {
+	switch harbor.HarborCluster.Spec.Storage.Kind {
+	case "azure":
+		return "azureSecret"
+	case "gcs":
+		return "gcsSecret"
+	case "swift":
+		return "swiftSecret"
+	case "s3":
+		return "s3Secret"
+	case "oss":
+		return "ossSecret"
+	}
+	return ""
+}
+
+// getStorageSecret will get a name of k8s secret which stores storage info
+func (harbor *HarborReconciler) getRegistryStorageSecret() string {
+	p := harbor.getStorageProperty("registrySecret")
+	if p == nil {
+		return ""
+	}
+	return p.ToString()
+}
+
+// getCoreDatabaseSecret will get a name of k8s secret which stores core-database info
+func (harbor *HarborReconciler) getCoreDatabaseSecret() string {
+	p := harbor.getDatabaseProperty("coreSecret")
+	if p == nil {
+		return ""
+	}
+	return ""
+}
+
+// getClairDatabaseSecret will get a name of k8s secret which stores clair-database info
+func (harbor *HarborReconciler) getClairDatabaseSecret() string {
+	p := harbor.getDatabaseProperty("clairSecret")
+	if p == nil {
+		return ""
+	}
+	return ""
+}
+
+// getNotaryServerDatabaseSecret will get a name of k8s secret which stores notary-server-database info
+func (harbor *HarborReconciler) getNotaryServerDatabaseSecret() string {
+	p := harbor.getDatabaseProperty("notaryServerSecret")
+	if p == nil {
+		return ""
+	}
+	return p.ToString()
+}
+
+// getClairDatabaseSecret will get a name of k8s secret which stores clair-database info
+func (harbor *HarborReconciler) getNotarySignerDatabaseSecret() string {
+	p := harbor.getDatabaseProperty("notarySignerSecret")
+	if p == nil {
+		return ""
+	}
+	return p.ToString()
+}
+
+func (harbor *HarborReconciler) getStorageProperty(name string) *lcm.Property {
+	return harbor.getProperty(goharborv1.ComponentStorage, name)
+}
+
+func (harbor *HarborReconciler) getDatabaseProperty(name string) *lcm.Property {
+	return harbor.getProperty(goharborv1.ComponentDatabase, name)
+}
+
+func (harbor *HarborReconciler) getCacheProperty(name string) *lcm.Property {
+	return harbor.getProperty(goharborv1.ComponentCache, name)
+}
+
+func (harbor *HarborReconciler) getProperty(component goharborv1.Component, name string) *lcm.Property {
+	if harbor.ComponentToCRStatus == nil {
+		return nil
+	}
+	crStatus := harbor.ComponentToCRStatus[component]
+	if crStatus == nil || len(crStatus.Properties) == 0 {
+		return nil
+	}
+	return crStatus.Properties.Get(name)
+}
+
+func (harbor *HarborReconciler) getImagePullSecrets() []corev1.LocalObjectReference {
+	if harbor.HarborCluster.Spec.ImageSource != nil && harbor.HarborCluster.Spec.ImageSource.ImagePullSecret != "" {
```

Comment:
It seems the `ImagePullSecret ` is not a mandatory property, you can double-check.

Reply:
Sorry, I don't understand what does this comment means?